samtranslator/plugins/globals/globals.py

```diff
@@ -111,12 +112,57 @@ def __init__(self, template) -> None:  # type: ignore[no-untyped-def]
         # Sort the names for stability in list ordering
         self.supported_resource_section_names.sort()
 
-        self.template_globals = {}
+        self.template_globals: Dict[str, GlobalProperties] = {}
 
         if self._KEYWORD in template:
             self.template_globals = self._parse(template[self._KEYWORD])  # type: ignore[no-untyped-call]
 
-    def merge(self, resource_type, resource_properties):  # type: ignore[no-untyped-def]
+    def get_template_globals(
+        self, logical_id: str, resource_type: str, ignore_globals: Optional[Union[str, List[str]]]
+    ) -> "GlobalProperties":
+        """
+        Get template globals but remove globals based on IgnoreGlobals attribute.
+
+        :param string logical_id: LogicalId of the resource
+        :param string resource_type: Type of the resource (Ex: AWS::Serverless::Function)
+        :param dict ignore_globals: IgnoreGlobals resource attribute. It can be either 1) "*" string value
+            or list of string value, each value should be a valid property in Globals section
+        :return dict: processed template globals
+        """
+        if not ignore_globals:
+            return self.template_globals[resource_type]
+
+        if isinstance(ignore_globals, str) and ignore_globals == "*":
+            return GlobalProperties({})
+
+        if isinstance(ignore_globals, list):
+            global_props: GlobalProperties = copy.deepcopy(self.template_globals[resource_type])
```

Comment:
Why do we need a deepcopy here?

Reply:
We need a copy because this attribute is resource level. We don't want to actually remove any item from the globals for other resources.